File: Aplikacja_Mobilna/App_IK/app/src/main/java/inteligenty_zamek/app_ik/CertificationaskActivity.java
package inteligenty_zamek.app_ik;

import android.content.res.TypedArray;
import android.os.AsyncTask;
import android.os.Bundle;
import android.os.CountDownTimer;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.SimpleAdapter;
import android.widget.Toast;

import org.apache.http.HttpEntity;
import org.apache.http.HttpResponse;
import org.apache.http.NameValuePair;
import org.apache.http.client.ClientProtocolException;
import org.apache.http.client.HttpClient;
import org.apache.http.client.entity.UrlEncodedFormEntity;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.impl.client.DefaultHttpClient;
import org.apache.http.message.BasicNameValuePair;
import org.apache.http.util.EntityUtils;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Iterator;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public class CertificationaskActivity extends BaseActivity {
    private String[] navMenuTitles;
    private TypedArray navMenuIcons;
    int toastDelay = 4000;
    JSONArray arrJson;
    List<LinkedHashMap<String, String>> listItems = new ArrayList<>();
    List<LinkedHashMap<String, String>> listItems2 = new ArrayList<>();

    ListView resultsListView;
    LinkedHashMap<String, String> Keys;
    String[][] locks;
    int size;
    LinkedHashMap<String, String> resultsMap;
CharSequence csk;
    SimpleAdapter adapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_certyficationask);
        navMenuTitles = getResources().getStringArray(R.array.nav_drawer_items);
        navMenuIcons = getResources().obtainTypedArray(R.array.nav_drawer_icons);
        set(navMenuTitles, navMenuIcons);

        try {
            User user = ((GlobalClassContainer) getApplication()).getUser();
            new CertificationaskActivity.HTTPRequestLocks(user).execute();
        } catch (Exception except) {
        }





       /* Log.i("aaaaaa",listItems.get(position).values().toArray()[0].toString());
        Log.i("bbbbb",locks[i][0]);
        Log.i("aaaaaa",listItems.get(position).values().toArray()[1].toString());
        Log.i("bbbbb",locks[i][1]);
*/



        EditText inputSearch = (EditText) findViewById(R.id.editText_Search);
        inputSearch.addTextChangedListener(new TextWatcher() {

            @Override
            public void onTextChanged(CharSequence cs, int arg1, int arg2, int arg3) {
                csk=cs;
                Keys = new LinkedHashMap<>();
                listItems2 = new ArrayList<>();
                adapter = new SimpleAdapter(CertificationaskActivity.this, listItems2, R.layout.main_key_list,
                        new String[]{"First Line", "Second Line"},
                        new int[]{R.id.TextView_liistNameKey, R.id.TextView_listPlaceKey});



                try {
                    for (int i = 0; i < listItems.size(); i++) {
                        Keys.put(listItems.get(i).values().toArray()[0].toString(), listItems.get(i).values().toArray()[1].toString());

                    }
                }catch (Exception e) {}
                Iterator it = Keys.entrySet().iterator();
                while (it.hasNext())
                {
                    resultsMap = new LinkedHashMap<>();
                    Map.Entry pair = (Map.Entry)it.next();
                    if(
                            pair.getKey().toString().toLowerCase().contains(cs.toString().toLowerCase())
                                    || pair.getValue().toString().toLowerCase().contains(cs.toString().toLowerCase())
                            ) {

                        resultsMap.put("First Line", pair.getKey().toString());
                        resultsMap.put("Second Line", pair.getValue().toString());
                        listItems2.add(resultsMap);
                    }
                    else if(cs.toString()=="")
                    {


                        resultsMap.put("First Line", pair.getKey().toString());
                        resultsMap.put("Second Line", pair.getValue().toString());
                        listItems2.add(resultsMap);
                    }
                }
                resultsListView.setAdapter(adapter);


                resultsListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
                    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                        int poz=-1;
                        Log.i("aaaaaaa","weszloo1");
                        for(int i=0; i<size; i++)
                        {
                            if(listItems2.get(position).values().toArray()[0].toString().equals(locks[i][0]) && listItems2.get(position).values().toArray()[1].toString().equals(locks[i][1]))
                            {poz=i; break;}
                        }
                        if(poz>=0) {
                            Log.i("aaaaaaa","weszloo");
                            String key_id = locks[poz][2];
                            User user = ((GlobalClassContainer) getApplication()).getUser();
                            new CertificationaskActivity.HTTPRequest(user, key_id).execute();
                        }
                    }
                });
            }

            @Override
            public void beforeTextChanged(CharSequence arg0, int arg1, int arg2,
                                          int arg3) {
                // TODO Auto-generated method stub

            }

            @Override
            public void afterTextChanged(Editable arg0) {
                // TODO Auto-generated method stub
            }
        });










    }

    public class HTTPRequestLocks extends AsyncTask<Void, Void, String> {
        User user;
        boolean choise;

        public HTTPRequestLocks(User x) {
            user = x;
        }

        @Override
        protected String doInBackground(Void... params) {
            HttpClient httpclient = new DefaultHttpClient();
            String adres = "http://" + ((GlobalClassContainer) getApplication()).getSerwerIP() + ":8080/api/download/all_locks/";
            HttpPost httppost = new HttpPost(adres);
            try {
                // Add your data
                List<NameValuePair> nameValuePairs = new ArrayList<NameValuePair>(2);
                nameValuePairs.add(new BasicNameValuePair("login", user.getLogin()));
                nameValuePairs.add(new BasicNameValuePair("token", ((GlobalClassContainer) getApplication()).getSession()));
                httppost.setEntity(new UrlEncodedFormEntity(nameValuePairs));
                // Execute HTTP Post Request
                HttpResponse response = httpclient.execute(httppost);
                HttpEntity entity = response.getEntity();
                String responseString = EntityUtils.toString(entity, "UTF-8");
                return responseString;
            } catch (ClientProtocolException e) {
                // TODO Auto-generated catch block
            } catch (IOException e) {
                // TODO Auto-generated catch block
            }
            return null;
        }

        //akcja po otrzyman iu odpowiedzi z serwera
        @Override
        protected void onPostExecute(String response) {
            super.onPostExecute(response);
            JSONObject jObj = null;
            try {
                jObj = new JSONObject(response);
                if (!jObj.getString("data").equals("empty")) {
                    arrJson = jObj.getJSONArray("data");
                    resultsListView = (ListView) CertificationaskActivity.this.findViewById(R.id.ListView_key_generate_certyfiaction);
                    //hasmap przechowujący elemety do wyświetlenia
                    Keys = new LinkedHashMap<>();
                    locks = new String[arrJson.length()][3];
                    size=arrJson.length();
                    for (int i = 0; i < arrJson.length(); i++) {
                        try {
                            locks[i][0] = arrJson.getJSONObject(i).getString("NAME");
                            locks[i][1] = arrJson.getJSONObject(i).getString("LOCALIZATION");
                            locks[i][2] = arrJson.getJSONObject(i).getString("ID_LOCK");
                            Keys.put(arrJson.getJSONObject(i).getString("NAME"), arrJson.getJSONObject(i).getString("LOCALIZATION"));
                        } catch (JSONException ignored) {
                        }
                    }

                    //stworzenie adaptera
                    adapter = new SimpleAdapter(CertificationaskActivity.this, listItems, R.layout.main_key_list,
                            new String[]{"First Line", "Second Line"},
                            new int[]{R.id.TextView_liistNameKey, R.id.TextView_listPlaceKey});

                    //iterator elementow (przepisanie z hashmap do adaptera[listitems] elementow)
                    Iterator it = Keys.entrySet().iterator();
                    while (it.hasNext()) {
                        resultsMap = new LinkedHashMap<>();
                        Map.Entry pair = (Map.Entry) it.next();
                        resultsMap.put("First Line", pair.getKey().toString());
                        resultsMap.put("Second Line", pair.getValue().toString());
                        listItems.add(resultsMap);
                    }
                    resultsListView.setAdapter(adapter);

                    resultsListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
                        public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                            //ustalenie id certyfikatu z listy certyfikatow
                            String key_id= locks[position][2];
                            User user = ((GlobalClassContainer) getApplication()).getUser();
                            new CertificationaskActivity.HTTPRequest(user, key_id).execute();
                        }
                    });

                    final Toast toast = Toast.makeText(CertificationaskActivity.this, "Pobrano listę zamków", Toast.LENGTH_LONG);
                    toast.show();
                    new CountDownTimer(toastDelay, 1000) {
                        public void onTick(long millisUntilFinished) {
                            toast.show();
                        }

                        public void onFinish() {
                            toast.show();
                        }
                    }.start();
                }
                else{
                    final Toast toast = Toast.makeText(CertificationaskActivity.this, "Lista pusta", Toast.LENGTH_LONG);
                    toast.show();
                    new CountDownTimer(toastDelay, 1000) {
                        public void onTick(long millisUntilFinished) {
                            toast.show();
                        }

                        public void onFinish() {
                            toast.show();
                        }
                    }.start();
                }
            } catch (JSONException e) {

                final Toast toast = Toast.makeText(CertificationaskActivity.this, "Niemożna pobrać listy zamków", Toast.LENGTH_LONG);
                toast.show();
                new CountDownTimer(toastDelay, 1000) {
                    public void onTick(long millisUntilFinished) {
                        toast.show();
                    }

                    public void onFinish() {
                        toast.show();
                    }
                }.start();
            }
        }
    }

    public class HTTPRequest extends AsyncTask<Void, Void, String> {
        User user;
        String id_lock;

        public HTTPRequest(User x, String id_lock) {
            user = x;
            this.id_lock = id_lock;
        }

        @Override
        protected String doInBackground(Void... params) {
            HttpClient httpclient = new DefaultHttpClient();
            String adres = "http://" + ((GlobalClassContainer) getApplication()).getSerwerIP() + ":8080/api/request_new_certificate/";
            HttpPost httppost = new HttpPost(adres);
            try {
                // Add your data
                List<NameValuePair> nameValuePairs = new ArrayList<NameValuePair>(3);
                nameValuePairs.add(new BasicNameValuePair("login", user.getLogin()));
                nameValuePairs.add(new BasicNameValuePair("token", ((GlobalClassContainer) getApplication()).getSession()));
                nameValuePairs.add(new BasicNameValuePair("lock_id", id_lock));
                httppost.setEntity(new UrlEncodedFormEntity(nameValuePairs));
                // Execute HTTP Post Request
                HttpResponse response = httpclient.execute(httppost);
                HttpEntity entity = response.getEntity();
                String responseString = EntityUtils.toString(entity, "UTF-8");
                return responseString;
            } catch (ClientProtocolException e) {
                // TODO Auto-generated catch block
            } catch (IOException e) {
                // TODO Auto-generated catch block
            }
            return null;
        }

        //akcja po otrzyman iu odpowiedzi z serwera
        @Override
        protected void onPostExecute(String response) {
            super.onPostExecute(response);
            JSONObject jObj = null;
            try {
                jObj = new JSONObject(response);
                if (jObj.getString("status").equals("ok")) {

                    final Toast toast = Toast.makeText(CertificationaskActivity.this, "Wysłano wniosek", Toast.LENGTH_LONG);
                    toast.show();
                    new CountDownTimer(toastDelay, 1000) {
                        public void onTick(long millisUntilFinished) {
                            toast.show();
                        }

                        public void onFinish() {
                            toast.show();
                        }
                    }.start();
                }
                else{
                    final Toast toast = Toast.makeText(CertificationaskActivity.this, "Wniosek odrzucono", Toast.LENGTH_LONG);
                    toast.show();
                    new CountDownTimer(toastDelay, 1000) {
                        public void onTick(long millisUntilFinished) {
                            toast.show();
                        }

                        public void onFinish() {
                            toast.show();
                        }
                    }.start();
                }
            } catch (JSONException e) {

                final Toast toast = Toast.makeText(CertificationaskActivity.this, "Wniosek odrzucono", Toast.LENGTH_LONG);
                toast.show();
                new CountDownTimer(toastDelay, 1000) {
                    public void onTick(long millisUntilFinished) {
                        toast.show();
                    }

                    public void onFinish() {
                        toast.show();
                    }
                }.start();
            }
        }
    }
}

File: Aplikacja_Mobilna/App_IK/app/src/main/java/inteligenty_zamek/app_ik/sampledata/CertificationaskActivity.kt
package inteligenty_zamek.app_ik.sampledata

import android.content.res.TypedArray
import android.os.AsyncTask
import android.os.Bundle
import android.os.CountDownTimer
import android.text.Editable
import android.text.TextWatcher
import android.util.Log
import android.widget.AdapterView
import android.widget.EditText
import android.widget.ListView
import android.widget.SimpleAdapter
import android.widget.Toast

import org.apache.http.NameValuePair
import org.apache.http.client.ClientProtocolException
import org.apache.http.client.entity.UrlEncodedFormEntity
import org.apache.http.client.methods.HttpPost
import org.apache.http.impl.client.DefaultHttpClient
import org.apache.http.message.BasicNameValuePair
import org.apache.http.util.EntityUtils
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

import java.io.IOException
import java.util.ArrayList
import java.util.LinkedHashMap

import inteligenty_zamek.app_ik.rest_class.GlobalClassContainer
import inteligenty_zamek.app_ik.rest_class.User
import inteligenty_zamek.app_ik.Navigation.BaseActivity
import inteligenty_zamek.app_ik.R
import java.util.Map

class CertificationaskActivity : BaseActivity() {
    private var navMenuTitles: Array<String>? = null
    private var navMenuIcons: TypedArray? = null
    internal var toastDelay = 4000
    private var arrJson: JSONArray?=null
    private var listItems: MutableList<LinkedHashMap<String, String>> = ArrayList()
    private var listItems2: MutableList<LinkedHashMap<String, String>> = ArrayList()

    private var resultsListView: ListView?=null
    private var Keys: LinkedHashMap<String, String>?=null
    private var locks: Array<Array<String>>?=null
    private var size: Int = 0
    private var resultsMap: LinkedHashMap<String, String>?=null
    private var csk: CharSequence?=null
    private var adapter: SimpleAdapter?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
      /*  setContentView(R.layout.activity_certyficationask)
        navMenuTitles = resources.getStringArray(R.array.nav_drawer_items)
        navMenuIcons = resources.obtainTypedArray(R.array.nav_drawer_icons)
        set(navMenuTitles, navMenuIcons)

        try {
            val user = (application as GlobalClassContainer).user
            CertificationaskActivity.HTTPRequestLocks(user).execute()
        } catch (except: Exception) {
        }




        val inputSearch = findViewById(R.id.editText_Search) as EditText
        inputSearch.addTextChangedListener(object : TextWatcher {

            override fun onTextChanged(cs: CharSequence, arg1: Int, arg2: Int, arg3: Int) {
                csk = cs
                Keys = LinkedHashMap()
                listItems2 = ArrayList()
                adapter = SimpleAdapter(this@CertificationaskActivity, listItems2, R.layout.main_key_list,
                        arrayOf("First Line", "Second Line"),
                        intArrayOf(R.id.TextView_listPlaceKey, R.id.TextView_listNameKey))



                try {
                    for (i in listItems.indices) {
                        Keys!!.put(listItems[i].values.toTypedArray()[0].toString(), listItems[i].values.toTypedArray()[1].toString())

                    }
                } catch (e: Exception) {
                }

                val it = Keys!!.entries.iterator()
                while (it.hasNext()) {
                    resultsMap = LinkedHashMap()
                    val pair = it.next() as Map.Entry<*, *>
                    if (pair.key.toString().toLowerCase().contains(cs.toString().toLowerCase()) || pair.value.toString().toLowerCase().contains(cs.toString().toLowerCase())) {

                        resultsMap!!.put("First Line", pair.key.toString())
                        resultsMap!!.put("Second Line", pair.value.toString())
                        listItems2.add(resultsMap!!)
                    } else if (cs.toString() === "") {


                        resultsMap!!.put("First Line", pair.key.toString())
                        resultsMap!!.put("Second Line", pair.value.toString())
                        listItems2.add(resultsMap!!)
                    }
                }
                resultsListView!!.adapter = adapter


                resultsListView!!.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
                    var poz = -1
                    for (i in 0 until size) {
                        if (listItems2[position].values.toTypedArray()[0].toString() == locks!![i][0] && listItems2[position].values.toTypedArray()[1].toString() == locks!![i][1]) {
                            poz = i
                            break
                        }
                    }
                    if (poz >= 0) {
                        val key_id = locks!![poz][2]
                        val user = (application as GlobalClassContainer).user
                        CertificationaskActivity.HTTPRequest(user, key_id).execute()
                    }
                }
            }

            override fun beforeTextChanged(arg0: CharSequence, arg1: Int, arg2: Int,
                                           arg3: Int) {
                // TODO Auto-generated method stub

            }

            override fun afterTextChanged(arg0: Editable) {
                // TODO Auto-generated method stub
            }
        })


    }

    inner class HTTPRequestLocks(internal var user: User) : AsyncTask<Void, Void, String>() {
        internal var choise: Boolean = false

        override fun doInBackground(vararg params: Void): String? {
            val httpclient = DefaultHttpClient()
            val adres = "http://" + (application as GlobalClassContainer).serwerIP + ":8080/api/download/all_locks/"
            val httppost = HttpPost(adres)
            try {
                // Add your data
                val nameValuePairs = ArrayList<NameValuePair>(2)
                nameValuePairs.add(BasicNameValuePair("login", user.login))
                nameValuePairs.add(BasicNameValuePair("token", (application as GlobalClassContainer).session))
                httppost.entity = UrlEncodedFormEntity(nameValuePairs)
                // Execute HTTP Post Request
                val response = httpclient.execute(httppost)
                val entity = response.entity
                return EntityUtils.toString(entity, "UTF-8")
            } catch (e: ClientProtocolException) {
                // TODO Auto-generated catch block
            } catch (e: IOException) {
                // TODO Auto-generated catch block
            }

            return null
        }

        //akcja po otrzyman iu odpowiedzi z serwera
        override fun onPostExecute(response: String) {
            super.onPostExecute(response)
            var jObj: JSONObject? = null
            try {
                jObj = JSONObject(response)
                if (jObj.getString("data") != "empty") {
                    arrJson = jObj.getJSONArray("data")
                    resultsListView = this@CertificationaskActivity.findViewById(R.id.ListView_key_generate_certyfiaction) as ListView
                    //hasmap przechowujący elemety do wyświetlenia
                    Keys = LinkedHashMap()
                    locks = Array(arrJson!!.length()) { arrayOfNulls(3) }
                    size = arrJson!!.length()
                    for (i in 0 until arrJson!!.length()) {
                        try {
                            locks!![i][0] = arrJson!!.getJSONObject(i).getString("NAME")
                            locks!![i][1] = arrJson!!.getJSONObject(i).getString("LOCALIZATION")
                            locks!![i][2] = arrJson!!.getJSONObject(i).getString("ID_LOCK")
                            Keys!!.put(arrJson!!.getJSONObject(i).getString("NAME"), arrJson!!.getJSONObject(i).getString("LOCALIZATION"))
                        } catch (ignored: JSONException) {
                        }

                    }

                    //stworzenie adaptera
                    adapter = SimpleAdapter(this@CertificationaskActivity, listItems, R.layout.main_key_list,
                            arrayOf("First Line", "Second Line"),
                            intArrayOf(R.id.TextView_listPlaceKey, R.id.TextView_listNameKey))

                    //iterator elementow (przepisanie z hashmap do adaptera[listitems] elementow)
                    val it = Keys!!.entries.iterator()
                    while (it.hasNext()) {
                        resultsMap = LinkedHashMap()
                        val pair = it.next() as Map.Entry<*, *>
                        Log.i("aaaaa", pair.key.toString())
                        resultsMap!!.put("First Line", pair.key.toString())
                        resultsMap!!.put("Second Line", pair.value.toString())
                        listItems.add(resultsMap!!)
                    }
                    resultsListView!!.adapter = adapter

                    resultsListView!!.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
                        //ustalenie id certyfikatu z listy certyfikatow
                        val key_id = locks!![position][2]
                        val user = (application as GlobalClassContainer).user
                        CertificationaskActivity.HTTPRequest(user, key_id).execute()
                    }

                    val toast = Toast.makeText(this@CertificationaskActivity, "Pobrano listę zamków", Toast.LENGTH_LONG)
                    toast.show()
                    object : CountDownTimer(toastDelay.toLong(), 1000) {
                        override fun onTick(millisUntilFinished: Long) {
                            toast.show()
                        }

                        override fun onFinish() {
                            toast.show()
                        }
                    }.start()
                } else {
                    val toast = Toast.makeText(this@CertificationaskActivity, "Lista pusta", Toast.LENGTH_LONG)
                    toast.show()
                    object : CountDownTimer(toastDelay.toLong(), 1000) {
                        override fun onTick(millisUntilFinished: Long) {
                            toast.show()
                        }

                        override fun onFinish() {
                            toast.show()
                        }
                    }.start()
                }
            } catch (e: JSONException) {

                val toast = Toast.makeText(this@CertificationaskActivity, "Niemożna pobrać listy zamków", Toast.LENGTH_LONG)
                toast.show()
                object : CountDownTimer(toastDelay.toLong(), 1000) {
                    override fun onTick(millisUntilFinished: Long) {
                        toast.show()
                    }

                    override fun onFinish() {
                        toast.show()
                    }
                }.start()
            }

        }
    }

    inner class HTTPRequest(internal var user: User, internal var id_lock: String) : AsyncTask<Void, Void, String>() {

        override fun doInBackground(vararg params: Void): String? {
            val httpclient = DefaultHttpClient()
            val adres = "http://" + (application as GlobalClassContainer).serwerIP + ":8080/api/request_new_certificate/"
            val httppost = HttpPost(adres)
            try {
                // Add your data
                val nameValuePairs = ArrayList<NameValuePair>(3)
                nameValuePairs.add(BasicNameValuePair("login", user.login))
                nameValuePairs.add(BasicNameValuePair("token", (application as GlobalClassContainer).session))
                nameValuePairs.add(BasicNameValuePair("lock_id", id_lock))
                httppost.entity = UrlEncodedFormEntity(nameValuePairs)
                // Execute HTTP Post Request
                val response = httpclient.execute(httppost)
                val entity = response.entity
                return EntityUtils.toString(entity, "UTF-8")
            } catch (e: ClientProtocolException) {
                // TODO Auto-generated catch block
            } catch (e: IOException) {
                // TODO Auto-generated catch block
            }

            return null
        }

        //akcja po otrzyman iu odpowiedzi z serwera
        override fun onPostExecute(response: String) {
            super.onPostExecute(response)
            var jObj: JSONObject? = null
            try {
                jObj = JSONObject(response)
                if (jObj.getString("status") == "ok") {

                    val toast = Toast.makeText(this@CertificationaskActivity, "Wysłano wniosek", Toast.LENGTH_LONG)
                    toast.show()
                    object : CountDownTimer(toastDelay.toLong(), 1000) {
                        override fun onTick(millisUntilFinished: Long) {
                            toast.show()
                        }

                        override fun onFinish() {
                            toast.show()
                        }
                    }.start()
                } else {
                    val toast = Toast.makeText(this@CertificationaskActivity, "Wniosek odrzucono", Toast.LENGTH_LONG)
                    toast.show()
                    object : CountDownTimer(toastDelay.toLong(), 1000) {
                        override fun onTick(millisUntilFinished: Long) {
                            toast.show()
                        }

                        override fun onFinish() {
                            toast.show()
                        }
                    }.start()
                }
            } catch (e: JSONException) {

                val toast = Toast.makeText(this@CertificationaskActivity, "Wniosek odrzucono", Toast.LENGTH_LONG)
                toast.show()
                object : CountDownTimer(toastDelay.toLong(), 1000) {
                    override fun onTick(millisUntilFinished: Long) {
                        toast.show()
                    }

                    override fun onFinish() {
                        toast.show()
                    }
                }.start()
            }

        }*/
    }
}
